viewer-prototype/src/browser/trace-viewer/components/timegraph-output-component.tsx

```diff
@@ -104,30 +148,69 @@ export class TimegraphOutputComponent extends AbstractTreeOutputComponent<Timegr
 
     renderTree(): React.ReactNode {
         return <React.Fragment>
-            {this.state.timegraphTree.map(entry => {
+            {this.state.timegraphTree.map((entry, i) => {
                 if (entry.parentId !== -1) {
-                    return entry.labels[0] + '\n';
+                    return <p style={{ height: this.props.style.rowHeight, margin: 0 }} key={i}>
+                        {entry.labels[0] + '\n'}
+                    </p>
                 }
             })}
         </React.Fragment>;
     }
 
+
     renderChart(): React.ReactNode {
         return <React.Fragment>
+
             {this.state.outputStatus === ResponseStatus.COMPLETED ?
-                <div id='timegraph-main' className='ps__child--consume' onWheel={ev => { ev.preventDefault(); ev.stopPropagation(); }} >
+                <div id='timegraph-main' className='ps__child--consume' onWheel={ev => { ev.preventDefault(); ev.stopPropagation(); }}>
                     {this.renderTimeGraphContent()}
+                    {
+                        this.state.isClicked && this.state.isTooltipOpened &&
+                        <div id='tooltip-box' ref={this.tooltipRef} >
+                            <button id='close' onClick={ev => { this.closeTooltip() }}>x</button>
+                            {this.renderTooltip()}
```

Comment:
I would put the tooltip as its own react component. I think there's nothing there specific to the time graph, so the XY chart's tooltip could use the same tooltip component. So you could create a new react component, TooltipBox and send it props like `<TooltipBox onClose={this.closeTooltip} tooltip=this.TooltipObject/>`

From what I understand of react, we must not hesitate to have as many small component classes as required. And I think the tooltip one can make for an easy-ish stateless and re-usable function component. 

Here's a simple article explaining how to make a typescript functional react component: https://medium.com/@ethan_ikt/react-stateless-functional-component-with-typescript-ce5043466011 Maybe it does not apply directly, but there should be tons of examples out there.